### PR TITLE
Fix comments, typos and add property validation to Microsoft.Extensions.Logging.AzureAppServices

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/aspnet/Logging/issues/508 https://github.com/aspnet/Logging/issues/506 https://github.com/aspnet/Logging/issues/507

### CODE REVIEW

File: src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs [hunk 1]
```diff
@@ -11,44 +11,151 @@ namespace Microsoft.Extensions.Logging.AzureAppServices
...
         /// Gets or sets the last section of log blob name.
+        /// Defaults to <c>"applicationLog.txt"</c>.
         /// </summary>
-        public string BlobName { get; set; } = "applicationLog.txt";
+        public string BlobName
+        {
+            get { return _blobName; }
+            set
+            {
+                if (value == null)
```
Comment (moozzyk) [inline src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs]: Is empty string fine?

File: src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs [hunk 2]
```diff
@@ -11,44 +11,151 @@ namespace Microsoft.Extensions.Logging.AzureAppServices
...
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(BlobBatchSize)} must be positive.");
+                }
+                _blobBatchSize = value;
+            }
+        }
 
         /// <summary>
-        /// Gets or sets a time to wait between checking for blob log batches.
+        /// Gets or sets a time to wait between checking for blob log batches. Defaults to 5 seconds.
```
Comment (BrennanConroy) [inline src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs]: Newline and `<c></c>`?

File: src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs [hunk 3]
```diff
@@ -11,44 +11,151 @@ namespace Microsoft.Extensions.Logging.AzureAppServices
...
+        private int _retainedFileCountLimit = 2;
+        private int _fileSizeLimit = 10 * 1024 * 1024;
+        private string _blobName = "applicationLog.txt";
+        private TimeSpan? _fileFlushPeriod = TimeSpan.FromSeconds(1);
+        private int _backgroundQueueSize;
+
         /// <summary>
-        /// Gets or sets a strictly positive value representing the maximum log size in bytes. Once the log is full, no more message will be appended.
+        /// Gets or sets a strictly positive value representing the maximum log size in bytes.
+        /// Once the log is full, no more message will be appended.
```
Comment (BrennanConroy) [inline src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs]: `messages`

**Comment by BrennanConroy** [comment]
🚢 :it:

File: src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs [hunk 4]
```diff
@@ -11,44 +11,152 @@ namespace Microsoft.Extensions.Logging.AzureAppServices
...
         /// Gets or sets a message template describing the output messages.
+        /// Defaults to <c>"{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level}] {Message}{NewLine}{Exception}"</c>.
         /// </summary>
-        public string OutputTemplate { get; set; } = "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level}] {Message}{NewLine}{Exception}";
+        public string OutputTemplate
+        {
+            get { return _outputTemplate; }
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs]: Argh, why are we checking for whitespace??

File: src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs [hunk 5]
```diff
@@ -11,44 +11,152 @@ namespace Microsoft.Extensions.Logging.AzureAppServices
...
         /// Gets or sets the last section of log blob name.
+        /// Defaults to <c>"applicationLog.txt"</c>.
         /// </summary>
-        public string BlobName { get; set; } = "applicationLog.txt";
+        public string BlobName
+        {
+            get { return _blobName; }
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs]: Ditto.

File: src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs [hunk 6]
```diff
@@ -11,44 +11,152 @@ namespace Microsoft.Extensions.Logging.AzureAppServices
...
         /// </summary>
-        public int BackgroundQueueSize { get; set; }
+        public int BackgroundQueueSize
+        {
+            get { return _backgroundQueueSize; }
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(BackgroundQueueSize)} must be positive or 0.");
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.AzureAppServices/AzureAppServicesDiagnosticsSettings.cs]: `... must be non-negative.`

**Pull request merged**